datalad/support/annexrepo.py

```diff
@@ -1818,25 +1829,28 @@ def drop(self, files, options=None, key=False):
             raise InsufficientArgumentsError("drop() requires at least to "
                                              "specify 'files' or 'options'")
 
-        options = options[:] if options else []
-        files = files[:] if files else []
+        options = assure_list(options)
+        files = assure_list(files)
 
-        output_lines = []
         if key:
             # we can't drop multiple in 1 line, and there is no --batch yet, so
             # one at a time
             options = options + ['--key']
-            for k in files:
-                std_out, std_err = \
-                    self._run_annex_command('drop', annex_options=options + [k])
-                output_lines.extend(std_out.splitlines())
+            res = [self._run_annex_command_json(
+                'drop',
+                args=options + [k],
+                jobs=jobs)
+                for k in files]
+            # `normalize_paths` ... magic, useful?
```

Comment:
Ah, okay. The diff looked like you were introducing it.
So - it can co go away, I think. `match_return_type` for `normalize_paths` is `True` by default, therefore the return type manipulation in here should be redundant.